risc0/zkvm/platform/src/syscall.rs

```diff
@@ -785,3 +787,45 @@ pub unsafe extern "C" fn sys_verify_integrity(
 extern "C" {
     pub fn sys_alloc_aligned(nwords: usize, align: usize) -> *mut u8;
 }
+
+/// `sys_fork()` creates a new process by duplicating the calling process. The
+/// new process is referred to as the child process. The calling process is
+/// referred to as the parent process.
+///
+/// The child process and the parent process run in separate memory spaces. At
+/// the time of `sys_fork()` both memory spaces have the same content.
+///
+/// # Return Value
+///
+/// On success, the PID of the child process (1) is returned in the parent, and
+/// 0 is returned in the child. On failure, -1 is returned in the parent, no
+/// child process is created.
```

Comment:
Why return on failure, rather than simply ending execution? In the case of an honest prover, it's unclear when you'd want to continue. In the case of a dishonest prover, success and failure don't mean very much.

Reply:
Because this is the POSIX spec (I copied all this from the man page)